File: leet_code/math/67/67.dart
class Solution {
  String addBinary(String a, String b) {
    String result = '';
    int carry = 0;
    
    int maxLen = a.length > b.length ? a.length : b.length;
    a = a.padLeft(maxLen, '0');
    b = b.padLeft(maxLen, '0');
    
    for (int i = maxLen - 1; i >= 0; i--) {
      int sumBits = carry + int.parse(a[i]) + int.parse(b[i]);
      result = (sumBits % 2).toString() + result; 
      carry = sumBits ~/ 2;  
    }
    
    if (carry != 0) {
      result = '1' + result;
    }
    
    return result;
  }
}

File: leet_code/math/67/67.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        result = []
        carry = 0
        
        max_len = max(len(a), len(b))
        a = a.zfill(max_len)
        b = b.zfill(max_len)
        
        
        for i in range(max_len - 1, -1, -1):
           
            sum_bits = carry + int(a[i]) + int(b[i])
            result.append(str(sum_bits % 2)) 
            carry = sum_bits // 2 
        
      
        if carry:
            result.append('1')
        
        
        return ''.join(result[::-1])
